src/service/action-impl.js

```diff
@@ -86,8 +87,26 @@ export class ActionService {
     /** @const {!Window} */
     this.win = win;
 
+    /** @const @private {!Object<string, function(!ActionInvocation)>} */
+    this.actionHandlers_ = {};
+
+    /** @param {!Vsync} */
+    this.vsync_ = vsyncFor(this.win);
+
     // Add core events.
     this.addEvent('tap');
+
+    // Add core actions.
+    this.addAction('hide', invocation => {
```

Comment:
I'm very worried that this will be used to make annoying ads. At the very least I'd like us to deny the action if the target elements contains `amp-ad`.